src/components/NetworkGraph.vue

```diff
@@ -120,8 +128,14 @@ export default {
   },
   methods: {
     nodeButtonClicked: function () {
+      this.$refs.overlay.style.display = 'block'
```

Comment:
Can you put the loader div inside the overlay div and only toggle the visiblility for one of them.